core/src/main/java/org/apache/brooklyn/core/mgmt/entitlement/WebEntitlementContext.java

```diff
@@ -38,16 +47,25 @@
     final String requestUniqueIdentifier;
     
     public WebEntitlementContext(String user, String sourceIp, String requestUri, String requestUniqueIdentifier) {
+        this(user, sourceIp, requestUri, requestUniqueIdentifier, new HashMap<>());
+    }
+
+    public WebEntitlementContext(String user, String sourceIp, String requestUri, String requestUniqueIdentifier, Map<String, Object> attributes) {
         this.user = user;
         this.sourceIp = sourceIp;
         this.requestUri = requestUri;
         this.requestUniqueIdentifier = requestUniqueIdentifier;
+        if(attributes!=null)
+            this.attributes = attributes;
```

Comment:
`if (attributes != null) this.attributes.addAll(attributes)` -- and no need for `else` block now